redis/src/cluster_async/mod.rs

```diff
@@ -969,15 +969,29 @@ where
         // When we no longer need to support Rust versions < 1.67, remove fast_math and transition to the ilog2 function.
         let num_of_nodes_to_query =
             std::cmp::max(fast_math::log2_raw(num_of_nodes as f32) as usize, 1);
-        let (_, found_topology_hash) = calculate_topology_from_random_nodes(
+        let (res, failed_connections) = calculate_topology_from_random_nodes(
             &inner,
             num_of_nodes_to_query,
             &read_guard,
             DEFAULT_NUMBER_OF_REFRESH_SLOTS_RETRIES,
         )
-        .await?;
-        let change_found = read_guard.get_current_topology_hash() != found_topology_hash;
-        Ok(change_found)
+        .await;
+
+        if let Ok((_, found_topology_hash)) = res {
+            if read_guard.get_current_topology_hash() != found_topology_hash {
+                return Ok(true);
```

Comment:
why do we need to return result instead of just a boolean, if this function doesn't return any errors? it also means that we don't need to call it with `retry`